File: 2022-09-30-docker-network/backend/src/main/java/action/in/blog/todo/controller/TodoController.java
package action.in.blog.todo.controller;

import action.in.blog.todo.Todo;
import action.in.blog.todo.service.TodoService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TodoController {

    private final TodoService todoService;

    public TodoController(TodoService todoService) {
        this.todoService = todoService;
    }

    @PostMapping("/todo")
    public void addTodo(@RequestBody Todo todo) {
        todoService.addTodo(todo);
    }
}

File: 2024-08-07-aws-ecr-setup/action-in-blog/backend/src/main/kotlin/com/example/demo/controller/TodoController.kt
package com.example.demo.controller

import com.example.demo.domain.Todo
import com.example.demo.service.TodoService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class TodoController(
    private val todoService: TodoService,
) {

    @GetMapping("/todos")
    fun getTodos(): List<Todo> {
        return todoService.getTodos()
    }

    @PostMapping("/todos")
    fun addTodo(
        @RequestBody todo: Todo
    ) {
        todoService.createTodo(todo)
    }
}
